capeinfra/pipeline/data.py

```diff
@@ -257,31 +239,47 @@ def __init__(
                     args["raw_bucket"],
                     args["clean_bucket"],
                     args["script_bucket"],
-                    script_path,
+                    self.config["script"],
                 )
             ),
             srvc_policy_attach=None,
             opts=ResourceOptions(parent=self),
         )
 
+        default_args = deepcopy(default_args)
+        if (
+            "pymodules" in self.config
+            and "--additional-python-modules" not in default_args
```

Comment:
the default is no python modules, but when you run the join on an empty list it would set it to an empty string. Is that something we want? This just makes sure if there are no specified `pymodules` then that key wouldn't be entered into the `default_args`

Reply:
Yeah, i read it wrong :) Carry on yhen...